org.osgi.test.junit4/src/main/java/org/osgi/test/common/annotation/PropertiesConverter.java

```diff
@@ -39,9 +42,24 @@ private static Object toValue(Property entry) {
 
 		boolean primitive = entry.type()
 			.equals(Type.PrimitiveArray);
-		Object result = createArray(entry.scalar(), primitive, entry.value().length);
+		String[] value;
+		if (NOT_SET.equals(entry.systemProperty())) {
+			value = entry.value();
+		} else {
+			Properties sysProps = System.getProperties();
+			if (sysProps.containsKey(entry.systemProperty())) {
+				String prop = sysProps.getProperty(entry.systemProperty());
```

Comment:
```suggestion
				String prop = sysProps.getProperty(entry.systemProperty()).trim();
```

To elide leading and trailing whitespace.